### PR TITLE
Rebind overwrite dialog by adding check for modification time bias

### PR DESCRIPTION
**Pull request by vzhukovskii**
### What does this PR do?
This changes proposal rebinds default file overwrite dialog by adding an additional check for modification time bias.

The problem which this changes proposal fixes described in [followed comment](https://github.com/eclipse/che/issues/16751#issuecomment-692768138).

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16751

#### Release Notes
N/A

#### Docs PR
N/A

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: extensions/eclipse-che-theia-filesystem/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "keywords": [
+    "theia-extension"
+  ],
+  "version": "0.0.1",
+  "files": [
+    "lib",
+    "src"
+  ],
+  "dependencies": {
+    "@theia/filesystem": "1.4.0-next.89694867"
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-filesystem/package.json:12]: Why the specific version?
Normally, we depend on Theia `next` everywhere.
The specific version is already pinned in [`yarn.lock`](https://github.com/eclipse/che-theia/blob/503ca56a3c1a7e7c9ccab1f975b49e8ecd0ea993/yarn.lock#L2255-L2257).

Comment (vzhukovskii) [reply extensions/eclipse-che-theia-filesystem/package.json:12]: With the next tag there're new changes for the file system api may income. This value can be modified to `next` when che-theia will be aligned to theia's master head. Or is it okay, that we will use `next` version for the filesystem library?

Comment (azatsarynnyy) [reply extensions/eclipse-che-theia-filesystem/package.json:12]: I see. I missed that moment.

**Review by azatsarynnyy — APPROVED** [review]

**Comment by vzhukovskii** [comment]
[ci-build]

**Pull request merged**